pkg/util/pki/common.go

```diff
@@ -215,9 +218,9 @@ func sortAndDedupe(hosts []string) []string {
 func ControllerUserForCluster(cluster *v1beta1.KafkaCluster) *v1alpha1.KafkaUser {
 	return &v1alpha1.KafkaUser{
 		ObjectMeta: templates.ObjectMeta(
-			fmt.Sprintf(BrokerControllerFQDNTemplate,
-				fmt.Sprintf(BrokerControllerTemplate, cluster.Name), cluster.Namespace, cluster.Spec.GetKubernetesClusterDomain()),
-			LabelsForKafkaPKI(cluster.Name, cluster.Namespace), cluster,
+			util.TruncateStringByLen(fmt.Sprintf(BrokerControllerFQDNTemplate, fmt.Sprintf(BrokerControllerTemplate, cluster.Name), cluster.Namespace, cluster.Spec.GetKubernetesClusterDomain()), MaxCertManagerCNLen),
```

Comment:
Do we need to truncate the name of the created `KafkaUser` custom resource as well? Or only just the CommonName field in the created `Certificate` resource? I'm asking as for K8s resource names different rules apply (see https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names)

Reply:
We need to ensure the `name` field of the generated `KafkaUser` doesn't exceed the length limitation as well because the [`clusterCertificateForUser`](https://github.com/banzaicloud/koperator/blob/master/pkg/pki/certmanagerpki/certmanager_user.go#L146) function uses it as the `spec.commonName` while generating a `Certificate` object for the corresponding `KafkaUser`

In fact, it is the first error reported by the operator when our user reported this issue in #445

Reply:
In `clusterCertificateForUser` you could just perform the truncation just before passing it into `spec.commonName`